Composer/packages/client/src/components/Header.tsx

```diff
@@ -95,33 +121,110 @@ export const Header = () => {
+      );
+    } else {
+      setStartPanelText(formatMessage('Start all bots'));
+    }
+  }, [runningBots]);
 
   const onUpdateAvailableClick = useCallback(() => {
     setAppUpdateShowing(true);
   }, []);
 
+  function dismissStartPanelViewer() {
+    setStartPanelView(false);
+  }
+
+  function handleStartOrStopAll() {
+    if (areBotsStarted) {
+      stopAllBots();
+      dismissStartPanelViewer();
+      setBotsInBotProjectStarted(false);
+    } else {
+      startAllBots();
+      setStartPanelView(true);
+    }
+  }
+
   const showUpdateAvailableIcon = status === AppUpdaterStatus.UPDATE_AVAILABLE && !showing;
 
   return (
     <div css={headerContainer} role="banner">
-      <div css={logo}>
-        <img
-          alt={formatMessage('Composer Logo')}
-          aria-label={formatMessage('Composer Logo')}
-          src={composerIcon}
-          style={{ marginLeft: '9px' }}
-        />
-        <div css={headerTextContainer}>
-          <div css={title}>{formatMessage('Bot Framework Composer')}</div>
-          {projectName && (
-            <Fragment>
-              <div css={divider} />
-              <span css={botName}>{`${projectName} (${locale})`}</span>
-            </Fragment>
-          )}
-        </div>
+      <img
+        alt={formatMessage('Composer Logo')}
+        aria-label={formatMessage('Composer Logo')}
+        src={composerIcon}
+        style={{ marginLeft: '9px' }}
+      />
+      <div css={headerTextContainer}>
+        <div css={title}>{formatMessage('Bot Framework Composer')}</div>
+        {projectName && (
+          <Fragment>
+            <div css={divider} />
+            <span css={botName}>{`${projectName} (${locale})`}</span>
+          </Fragment>
+        )}
       </div>
-      <div css={controls}>
+
+      <div css={rightSection}>
+        {showStartBotsWidget && (
+          <div css={startBotWidgetContainer}>
+            {runningBots.projectIds.length > 0 ? (
+              <ActionButton
+                aria-label={formatMessage('Stop all bots')}
+                css={actionButton}
+                onClick={handleStartOrStopAll}
+              >
+                <Icon iconName={'CircleStopSolid'} styles={localBotOperationIcon} />
+              </ActionButton>
+            ) : (
+              <ActionButton
+                aria-label={formatMessage('Start all bots')}
+                css={actionButton}
+                data-testid={'startAllBots'}
+                onClick={handleStartOrStopAll}
+              >
+                <Icon iconName={'Play'} styles={localBotOperationIcon} />
+              </ActionButton>
+            )}
+            <span aria-label={startPanelText} aria-live={'assertive'}>
+              {startPanelText}
+            </span>
+            <ActionButton css={actionButton} onClick={() => setStartPanelView(true)}>
+              <Icon iconName={'ProductList'} styles={localBotOperationIcon} />
+            </ActionButton>
+          </div>
+        )}
+
```

Comment:
We should pull this out into a separate component to declutter the Header component. We can do that in my PR though.